### ISSUE TITLE
Multiple Training Areas within the Same Scene but result worse!

### ISSUE THREAD

**Issue by Kuriboh2017** [Unity-Technologies/ml-agents#5808]
I used Karting Microgame template to train the kart agent.
I used ml-agents release_19 and updated this Karting game.  
It worked well so I tried to copy 8 training areas to speed up trainig. Here is my check test.
![tensorboard](https://user-images.githubusercontent.com/60729226/198543705-296c09f9-e0be-4735-8c73-64fde4fab989.png)

**carmine line**: one agent with 1M steps.
**green line**: 10 agents with 5M steps. the result is worth than single agent.
**purple line**: 10 agents with 2 phase: 1M steps firstly, then i resume the training to 2M steps. Amazing, it reached single agent's level in second phase.
**orange line**: 10 agents with 3 phase:500k steps firstly, then resume to 1M steps secondly, resume to 2M steps finally. Now i am convinced that each time I resume the training it become sharply better than before. BUT Why?

Do I need to adjust the hyperparamaters because i didn't change the config.
Here is my configuration.

behaviors:
    ArcadeDriver:
        trainer_type: ppo
        hyperparameters:
            batch_size: 1024
            beta: 5.0e-3
            buffer_size: 10240
            epsilon: 0.2
            lambd: 0.95
            num_epoch: 3
            learning_rate: 3.0e-4
            learning_rate_schedule: linear
        network_settings:
            normalize: false
            hidden_units: 128           
            num_layers: 2
            vis_encode_type: simple
        reward_signals:
            extrinsic:
                strength: 1.0
                gamma: 0.99
        max_steps: 1000000
        time_horizon: 64
        summary_freq: 5000
